src/java/za/co/mip/ablduck/ABLDuckClassVisitor.java

```diff
@@ -0,0 +1,214 @@
+                    commentParseResult.internal = method.methodName.startsWith("_");
+
+                m.returnComment = commentParseResult.returnComment;
+                m.comment = commentParseResult.comment;
+
+                if (commentParseResult.internal)
+                    m.meta.internal = "This is a private method for internal use by the framework. Don't rely on its existence.";
+
+                //TODO: remove this duplication tsk tsk, write a meta adding method for class, method and property
+                if (!commentParseResult.deprecatedVersion.equals("")) {
+                    m.meta.deprecated = new DeprecatedObject();
+        
+                    m.meta.deprecated.version = commentParseResult.deprecatedVersion;
+                    m.meta.deprecated.text    = commentParseResult.deprecatedText;
+                }
+            } catch (IOException ex) {
+                throw ex;
+            }
+            
+            switch (method.modifier){
+                case PRIVATE:
+                    m.meta.private_ = true;
+                    break;
+                case PROTECTED:
+                    m.meta.protected_ = true;
+                    break;
+            }
+            
+            if (method.isAbstract)
+                m.meta.abstract_ = true;
+
+            if (method.isStatic)
+                m.meta.static_ = true;
+
+            m.parameters = new ArrayList<ParameterObject>();
+            for (Parameter parameter : method.parameters) {
+                ParameterObject p = new ParameterObject();
+
+                p.name = parameter.name;
+                p.datatype = parameter.dataType;
+
+                //TODO: do i need this? probably need to show this in the docs somewhere
+                p.mode = parameter.mode.toString();
+
+                m.parameters.add(p);
+            }
+
+            js.members.add(m);
+        }
+
+        //TODO: remove this duplication tsk tsk, write a meta adding method for class, method and property
+        //Properties
+        for (Property property : cu.properties) {
+            MemberObject m = new MemberObject();
+
+            m.id = "property-" + property.name;
+            m.name = property.name;
+            m.owner = fullClassName;
+            m.tagname = "property";
+            m.datatype = property.dataType;
+
+            try {
+                CommentParseResult commentParseResult = CommentParser.parseComment(property.propertyComment, fullClassName + ":" + property.name);
+
+                if (!commentParseResult.internal)
+                    commentParseResult.internal = property.name.startsWith("_");
+
+                m.comment = commentParseResult.comment;
+
+                if (commentParseResult.internal)
+                    m.meta.internal = "This is a private property for internal use by the framework. Don't rely on its existence.";
+
+                if (!commentParseResult.deprecatedVersion.equals("")) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the "" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1132)